ui/src/main/java/edu/wpi/grip/ui/codeGeneration/TemplateMethods.java

```diff
@@ -0,0 +1,130 @@
+package edu.wpi.grip.ui.codegeneration;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+import edu.wpi.grip.core.Connection;
+import edu.wpi.grip.core.Pipeline;
+import edu.wpi.grip.core.Step;
+import edu.wpi.grip.core.sockets.InputSocket;
+import edu.wpi.grip.core.sockets.Socket;
+import edu.wpi.grip.core.sockets.SocketHint;
+
+/**
+ * Created by Toby on 5/25/16.
+ */
+public class TemplateMethods {
+
+  private Pipeline pipeline;
+  private Map<Connection, String> connections;
+  private int numOutputs;
+  private int numSources;
+
+  public TemplateMethods() {
+    connections = new HashMap<Connection, String>();
+    numOutputs = 0;
+    numSources = 0;
+  }
+
+  public void setPipeline(Pipeline pipeline) {
+    this.pipeline = pipeline;
+  }
+
+  public static String parseSocketValue(Socket socket) {
+    String value = socket.getValue().toString();
+    if(socket.getSocketHint().getView().equals(SocketHint.View.NONE)){
+      return "null" + value;
+    }
+    else if (value.contains("Optional[")) {
+      return value.substring(value.indexOf("[") + 1, value.lastIndexOf("]"));
+    } else {
+      return value;
+    }
+  }
+
+  public static String parseSocketName(Socket socket) {
+    String name = socket.getSocketHint().getIdentifier();
+    return toCamelCase(name);
+  }
+
+  public static String toCamelCase(final String init) {
+    if (init == null)
+      return null;
+
+    final StringBuilder ret = new StringBuilder(init.length());
+
+    boolean firstWord = true;
+    for (final String word : init.split(" ")) {
+      if (firstWord) {
+        firstWord = false;
+        ret.append(word.substring(0).toLowerCase());
+      } else if (!word.isEmpty()) {
+        ret.append(word.substring(0, 1).toUpperCase());
+        ret.append(word.substring(1).toLowerCase());
+      }
+    }
+
+    return ret.toString();
+  }
+
+  public static String parseSocketType(Socket socket) {
+    String type = socket.getSocketHint().getType().getSimpleName();
+    if(socket.getSocketHint().getView().equals(SocketHint.View.SELECT)){
+      if(type.contains("BorderTypes") || type.contains("CmpTypes")){
```

Comment:
Use the actual class type for comparisons here.
So do, `CmpType.class.equals(socket.getSocketHint().getType())`